frontend/src/selectors/exerciseSelectors.js

```diff
@@ -17,16 +17,19 @@ export const getExerciseTimer = createSelector(
   exercise => exercise.timer
 );
 
-export const getThreads = createSelector(exerciseSelector, exercise =>
-  exercise.threads
-    .filter(thread => thread.revealTime <= exercise.timer)
-    .sort((a, b) => b.revealTime - a.revealTime)
+export const getThreads = createSelector(
+  exerciseSelector,
+  exercise =>
+    exercise.threads
+      .filter(thread => thread.threadProperties.revealTime <= exercise.timer)
+      .sort((a, b) => b.revealTime - a.revealTime)
```

Comment:
This one needs to be done also:
```suggestion
      .sort((a, b) => b.threadProperties.revealTime - a.threadProperties.revealTime)
```